flytekit/core/gate.py

```diff
@@ -0,0 +1,123 @@
+from __future__ import annotations
+
+import datetime
+import typing
+from typing import Tuple, Union
+
+from flytekit.core import interface as flyte_interface
+from flytekit.core.context_manager import FlyteContext
+from flytekit.core.promise import Promise, VoidPromise, flyte_entity_call_handler
+from flytekit.core.type_engine import TypeEngine
+from flytekit.loggers import logger
+from flytekit.models.core import workflow as _workflow_model
+from flytekit.models.types import LiteralType
+
+DEFAULT_TIMEOUT = datetime.timedelta(hours=1)
+
+
+class Gate(object):
+    """
+    A gate needs to behave kind of like a task. Instead of running code however, it either needs to wait for user
+    input, or run a timer.
```

Comment:
```suggestion
    A node type that waits for user input before proceeding with a workflow.

    A gate is a type of node that behaves like a task, but instead of running code, it either needs to wait for user input to proceed or wait for a timer to complete running.
```